src/pages/iou/steps/IOUAmountPage.js

```diff
@@ -65,6 +58,45 @@ class IOUAmountPage extends React.Component {
         }
     }
 
+    /**
+     * Update amount with number or Backspace pressed.
+     * Validate new amount with decimal number regex up to 6 digits and 2 decimal digit to enable Next button
+     *
+     * @param {String} key
+     * @param {String} event
+     */
+    onlyAllowNumericValues(key, event) {
+        // Prevent reusing synthetic event object, so we can call preventDefault() later.
+        if (event) {
+            event.persist();
+        }
+
+        // Backspace button is pressed
+        if (key === '<' || key === 'Backspace') {
+            if (this.state.amount.length > 0) {
+                this.setState(prevState => ({
+                    amount: prevState.amount.substring(0, prevState.amount.length - 1),
+                    isNextButtonEnabled: prevState.amount.length !== 1,
+                }));
+            }
+            return;
+        }
+        this.setState((prevState) => {
+            const newValue = `${prevState.amount}${key}`;
+            const decimalNumberRegex = new RegExp(/^\d{1,6}(\.\d{0,2})?$/, 'i');
```

Comment:
What is this regex testing for? Add a comment?

Reply:
I don't know if I missed this somewhere... but why is there a limit of 6 digits? Is that just ambiguous? Let's allow for any number of digits. Also with the decimals, some currencies allow three decimal places, so that needs to be updated too.

Reply:
Was your question resolved here @tgolen ?

Reply:
Sorry, I missed this comment earlier. I changed the regex pattern with the last commit.👍